controllers/testutil/equalities.go

```diff
@@ -0,0 +1,154 @@
+	expectedConfigMap := &v1.ConfigMap{}
+	actualConfigMap := &v1.ConfigMap{}
+	err := scheme.Scheme.Convert(expected, expectedConfigMap, nil)
+	if err != nil {
+		return false, err
+	}
+	err = scheme.Scheme.Convert(actual, actualConfigMap, nil)
+	if err != nil {
+		return false, err
+	}
+
+	if expectedConfigMap.Name != actualConfigMap.Name {
+		return false, notEqualMsg("Configmap Names are not equal.")
+	}
+
+	if !reflect.DeepEqual(expectedConfigMap.Data, actualConfigMap.Data) {
+		return false, notEqualMsg("Configmap's Data values")
+	}
+	return true, nil
+}
+
+func secretsAreEqual(expected, actual *unstructured.Unstructured) (bool, error) {
+	expectedSecret := &v1.Secret{}
+	actualSecret := &v1.Secret{}
+	err := scheme.Scheme.Convert(expected, expectedSecret, nil)
+	if err != nil {
+		return false, err
+	}
+	err = scheme.Scheme.Convert(actual, actualSecret, nil)
+	if err != nil {
+		return false, err
+	}
+
+	if expectedSecret.Name != actualSecret.Name {
+		return false, notEqualMsg("Secret Names are not equal.")
+	}
+
+	if !reflect.DeepEqual(expectedSecret.Data, actualSecret.Data) {
+		return false, notEqualMsg("Secret's Data values")
+	}
+	return true, nil
+}
+
+func deploymentsAreEqual(expected, actual *unstructured.Unstructured) (bool, error) {
+	expectedDep := &appsv1.Deployment{}
+	actualDep := &appsv1.Deployment{}
+	err := scheme.Scheme.Convert(expected, expectedDep, nil)
+	if err != nil {
+		return false, err
+	}
+	err = scheme.Scheme.Convert(actual, actualDep, nil)
+	if err != nil {
+		return false, err
+	}
+	if !reflect.DeepEqual(expectedDep.ObjectMeta.Labels, actualDep.ObjectMeta.Labels) {
+		return false, notEqualMsg("labels")
+	}
+
+	if !reflect.DeepEqual(expectedDep.Spec.Selector, actualDep.Spec.Selector) {
+		return false, notEqualMsg("selector")
+	}
+
+	if !reflect.DeepEqual(expectedDep.Spec.Template.ObjectMeta, actualDep.Spec.Template.ObjectMeta) {
+		return false, notEqualMsg("selector")
+	}
+
+	if !reflect.DeepEqual(expectedDep.Spec.Template.Spec.Volumes, actualDep.Spec.Template.Spec.Volumes) {
+		return false, notEqualMsg("Volumes")
+	}
+
+	if len(expectedDep.Spec.Template.Spec.Containers) != len(actualDep.Spec.Template.Spec.Containers) {
+		return false, notEqualMsg("Containers")
+	}
+	for i := range expectedDep.Spec.Template.Spec.Containers {
+		expectedContainer := expectedDep.Spec.Template.Spec.Containers[i]
+		actualContainer := actualDep.Spec.Template.Spec.Containers[i]
+
+		if len(expectedContainer.Env) != len(actualContainer.Env) {
+			return false, notEqualMsg("Container Env Lengths ")
+		}
+		// Check each env individually for a more meaningful response upon failure.
+		for i, expectedEnv := range expectedContainer.Env {
+			actualEnv := actualContainer.Env[i]
+			if !reflect.DeepEqual(expectedEnv, actualEnv) {
+				return false, notEqualMsg(fmt.Sprintf("Container Env [expected: %s, actual: %s]", expectedEnv.Name, actualEnv.Name))
+			}
+		}
+
+		if !reflect.DeepEqual(expectedContainer.Env, actualContainer.Env) {
```

Comment:
one question: if we are checking each env above part,
then is this checking for the same ?